index/segment/mem/iterator.go

```diff
@@ -34,11 +34,12 @@ var (
 type iterator struct {
 	segment      ReadableSegment
 	postingsIter postings.Iterator
-	maxID        postings.ID
 
 	current doc.Document
-	err     error
-	closed  bool
+
+	err    error
+	closed bool
+	maxID  postings.ID
```

Comment:
Is this because of structcheck? If so, is it worth preferring the old layout since it had the benefit of distinguishing immutable versus mutable fields.

Reply:
it's because of `maligned`: this layout is 80 bytes, the older one was 88. 

I'll put a comment to indicate why we order this way and the mutability of fields.

Reply:
done